rust/ql/lib/codeql/rust/internal/Type.qll

```diff
@@ -29,7 +29,26 @@ newtype TType =
 abstract class Type extends TType {
   /** Gets the method `name` belonging to this type, if any. */
   pragma[nomagic]
-  abstract Function getMethod(string name);
+  final Function getMethod(string name) {
+    result = this.getAMethod(name) and
+    (
+      // when a method exists in both source code and in library code, it is because
+      // we also extracted the source code as library code, and hence we only want
+      // the method from source code
```

Comment:
Does this not affect path resolution as well? Could the workaround be done in path resolution? Or is there a benefit to having `getASuccessor` give the duplicated methods?